File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/options/RemoveMarkersByIdsOptions.java
package io.capawesome.capacitorjs.plugins.maplibre.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.classes.CustomExceptions;
import java.util.List;

public class RemoveMarkersByIdsOptions {

    @NonNull
    private final String mapId;

    @NonNull
    private final List<String> markerIds;

    public RemoveMarkersByIdsOptions(@NonNull PluginCall call) throws Exception {
        JSArray markerIdsArray = call.getArray("markerIds", null);
        if (markerIdsArray == null) {
            throw CustomExceptions.MARKER_IDS_MISSING;
        }
        this.mapId = MapLibreHelper.getMapIdFromCall(call);
        this.markerIds = MapLibreHelper.createStringList(markerIdsArray);
    }

    @NonNull
    public String getMapId() {
        return mapId;
    }

    @NonNull
    public List<String> getMarkerIds() {
        return markerIds;
    }
}

File: packages/maplibre/ios/Plugin/Classes/Options/RemoveMarkersByIdsOptions.swift
import Capacitor
import Foundation

@objc public class RemoveMarkersByIdsOptions: NSObject {
    let mapId: String
    let markerIds: [String]

    init(_ call: CAPPluginCall) throws {
        self.mapId = try MapLibreHelper.getString(call, "mapId", .mapIdMissing)
        self.markerIds = try MapLibreHelper.getStringArray(call, "markerIds", .markerIdsMissing)
    }
}
